hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/QuotaUtil.java

```diff
@@ -321,6 +344,34 @@ public void visitUserQuotas(String userName, Quotas quotas) {
     return userQuotas;
   }
 
+  protected static UserQuotaState buildDefaultUserQuotaState(Configuration conf) {
+    QuotaProtos.Throttle.Builder throttleBuilder = QuotaProtos.Throttle.newBuilder();
+
+    buildDefaultTimedQuota(conf, QUOTA_DEFAULT_USER_MACHINE_READ_SIZE)
+      .ifPresent(throttleBuilder::setReadSize);
+    buildDefaultTimedQuota(conf, QUOTA_DEFAULT_USER_MACHINE_READ_NUM)
+      .ifPresent(throttleBuilder::setReadNum);
+    buildDefaultTimedQuota(conf, QUOTA_DEFAULT_USER_MACHINE_REQUEST_NUM)
+      .ifPresent(throttleBuilder::setReqNum);
+    buildDefaultTimedQuota(conf, QUOTA_DEFAULT_USER_MACHINE_WRITE_NUM)
+      .ifPresent(throttleBuilder::setWriteNum);
+
+    UserQuotaState state = new UserQuotaState();
+    QuotaProtos.Quotas defaultQuotas =
+      QuotaProtos.Quotas.newBuilder().setThrottle(throttleBuilder.build()).build();
+    state.setQuotas(defaultQuotas);
+    return state;
+  }
+
+  private static Optional<TimedQuota> buildDefaultTimedQuota(Configuration conf, String key) {
+    int defaultSoftLimit = conf.getInt(key, -1);
+    if (defaultSoftLimit == -1) {
+      return Optional.empty();
+    }
+    return Optional.of(ProtobufUtil.toTimedQuota(defaultSoftLimit,
+      java.util.concurrent.TimeUnit.SECONDS, org.apache.hadoop.hbase.quotas.QuotaScope.MACHINE));
```

Comment:
Is there a reason we need these fully qualified names instead of importing?

Reply:
Fair question, but sadly the answer is yes because we already import
`import org.apache.hadoop.hbase.shaded.protobuf.generated.HBaseProtos.TimeUnit;`
and
`import org.apache.hadoop.hbase.shaded.protobuf.generated.QuotaProtos.QuotaScope;`